### PR TITLE
feat(glossary-item): use react-tippy to control tooltips

### PR DESCRIPTION
**Pull request by kevinports**
[![PR App][icn]][demo] | Fix RM-4416
:-------------------:|:----------:

The glossary item tooltip gets clipped inside of containers without visible overflow: 

image

This is a downside of pure css tooltips, and the solution is to use a JS controlled tooltip that can insert the popover outside of the clipping container.

## 🧰 Changes
- Add `react-tippy` as a peer dependency, because we also use it in the `readme` project's react package.
- Use tippy for the `GlossaryItem` component's tooltip, and append the popover element to `document.body`
- Update some docs

## 🧬 QA & Testing

- [Broken on production][prod].
- [Working in this PR app][demo].

[demo]: https://markdown-pr-553.herokuapp.com/#features
[prod]: https://SUBDOMAIN.readme.io
[icn]: https://user-images.githubusercontent.com/886627/160426047-1bee9488-305a-4145-bb2b-09d8b757d38a.svg

### CODE REVIEW

File: example/Demo.jsx [hunk 1]
```diff
@@ -15,6 +15,14 @@ const terms = [
...
     definition: 'a thing that breaks on presentation',
   },
+  {
+    term: 'exogenous',
+    definition: 'relating to or developing from external factors',
+  },
+  {
+    term: 'endogenous',
+    definition: 'having an internal cause or origin',
+  },
```
Comment (kevinports) [inline example/Demo.jsx:25]: Added so the glossary terms actually get tooltips in the demo here: https://markdown-pr-553.herokuapp.com/#features

**Comment by kevinports** [comment]
@kellyjosephprice I'm not really sure how to update the failing snapshots from `npm run test.browser`. It looks like the `__tests__/browser/__image_snapshots__` is included in the `.gitignore` file.

File: package.json [hunk 2]
```diff
@@ -56,6 +56,7 @@
   },
   "peerDependencies": {
     "@readme/variable": "^15.1.0",
+    "@tippyjs/react": "^4.1.0",
```
[the remark this answers is not in the document]
Comment (kevinports) [reply package.json:59]: Yep, added.

**Comment by kellyjosephprice** [comment]
> @kellyjosephprice I'm not really sure how to update the failing snapshots from `npm run test.browser`. It looks like the `__tests__/browser/__image_snapshots__` is included in the `.gitignore` file.

Yea, it's really obnoxious, sorry. Since we weren't able to get everyone's env's to be equivalent for the browser tests, we're relying on CI to generate the snapshots.

Here's a bit from the [docs](https://github.com/readmeio/markdown/blob/next/CONTRIBUTING.md#visual-regression-tests). You can download the diffs and snapshots [here](https://github.com/readmeio/markdown/actions/runs/2747897102). I think they live at `__tests__/browser/ci/`.

**Comment by kevinports** [comment]
Ah didn't see that bit in the docs. Thanks for clarifying!

**Review by kellyjosephprice — APPROVED** [review]
LGTM!

Is the snapshot update just that you added the definitions?! That's annoying.

**Review by erunion — APPROVED** [review]

**Comment by kevinports** [comment]
> Is the snapshot update just that you added the definitions?! That's annoying.

Yeah it is haha. Maybe a little annoying, but cool that we actually have some visual snapshot tests running!

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v6.48.0</b>](/readmeio/markdown/releases/tag/v6.48.0)
<br>

**Pull request merged**